src/Agent/NewRelic/Agent/Core/DataTransport/IHttpClientFactory.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+
+namespace NewRelic.Agent.Core.DataTransport
+{
+    public interface IHttpClientFactory
```

Comment:
New guy ignorance time: Why are we using a home-grown `IHttpClientFactory` rather than the built-in `Microsoft.Extensions.Http.IHttpClientFactory`?

Reply:
Mostly I tried to avoid referencing external assemblies as much as possible so that we didn't have to ILRepack.